ote/src/clj/ote/services/transit_changes.clj

```diff
@@ -106,37 +106,50 @@
   (GET "/transit-changes/current" []
        (#'list-current-changes db))
 
+  (GET "/transit-changes/hash-recalculations" {user :user :as request}
+       (when (authorization/admin? user)
+         (http/transit-response (detection/hash-recalculations db))))
+
+  ;; Calculate date-hashes for given service-id and package-count
   (GET "/transit-changes/force-calculate-hashes/:service-id/:package-count" [service-id package-count :as {user :user}]
     (when (authorization/admin? user)
-      (detection/calculate-package-hashes-for-service db (Long/parseLong service-id) (Long/parseLong package-count))
+      (detection/calculate-package-hashes-for-service db (Long/parseLong service-id) (Long/parseLong package-count) (:user user))
       "OK"))
 
+  ;; Calculate route-hash-id for given service-id and package-count
   (GET "/transit-changes/force-calculate-route-hash-id/:service-id/:package-count/:type" [service-id package-count type :as {user :user}]
     (when (authorization/admin? user)
         (detection/calculate-route-hash-id-for-service db (Long/parseLong service-id) (Long/parseLong package-count) type)
         "OK"))
 
-  (GET "/transit-changes/force-monthly-day-hash-calculation" {user :user}
+  ;; Calculate date-hashes for services using latest monthly package
+  (GET "/transit-changes/force-monthly-day-hash-calculation/:future" [future :as {user :user}]
     (when (authorization/admin? user)
       (http/transit-response
         {:status 200
-         :body (detection/calculate-monthly-date-hashes-for-packages db)})))
+         :body (detection/calculate-monthly-date-hashes-for-packages
+                 db (:user user) (if (= "true" future) true false))})))
```

Comment:
Doesn't (= "true" future) return true / false without the outer if?

Reply:
Nice! yes it does.